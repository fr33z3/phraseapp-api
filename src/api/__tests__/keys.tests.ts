import {addTagsToKeyCollection, createKey, deleteKey, deleteKeyCollection, excludeLocaleOnKeyCollection, getKey, includeLocaleOnKeyCollection, listKeys, removeTagsFromKeyCollection, searchKeys, updateKey} from 'api/keys'

describe('Key api', () => {
  const projectId = '77617b5199af590fde8642fb4178eaa1'

  describe('listKeys', () => {
    it('returns all keys for a project', async () => {
      const result = await listKeys(projectId)
      expect(result).toMatchSnapshot()
    })
  })

  describe('getKey', () => {
    it('returns all keys for a project', async () => {
      const result = await getKey(projectId, 'cd0764df095295dccf8e5a0ab7c32e17')
      expect(result).toMatchSnapshot()
    })
  })

  describe('createKey', () => {
    it('creates key and return key data', async () => {
      const result = await createKey(projectId, {
        name: 'test.key',
      })
      expect(result).toMatchSnapshot()
    })
  })

  describe('updateKey', () => {
    it('updates key and return key data', async () => {
      const result = await updateKey(projectId, 'fa22d14822b4f6bc2660d110729194bf', {
        name: 'test.key-changed-name',
      })
      expect(result).toMatchSnapshot()
    })
  })

  describe('deleteKey', () => {
    it('deletes key', async () => {
      const result = await deleteKey(projectId, 'fa22d14822b4f6bc2660d110729194bf')
      expect(result).toBeTruthy()
    })
  })

  describe('searchKeys', () => {
    it('searches keys by given query criteria', async () => {
      const result = await searchKeys(projectId, {q: 'general.*'})
      expect(result).toMatchSnapshot()
    })
  })

  describe('addTagsToKeyCollection', () => {
    it('adds tags to a collection of keys and return number of affected keys', async () => {
      const result = await addTagsToKeyCollection(projectId, {q: 'general.*', tags: ['tag1', 'tag2']})
      expect(result).toMatchInlineSnapshot(`
        Object {
          "records_affected": 6,
        }
      `)
    })
  })

  describe('removeTagsFromKeyCollection', () => {
    it('removes tags from a collection of keys and return number of affected keys', async () => {
      const result = await removeTagsFromKeyCollection(projectId, {q: 'general.*', tags: ['tag1']})
      expect(result).toMatchInlineSnapshot(`
        Object {
          "records_affected": 6,
        }
      `)
    })
  })

  describe('includeLocaleOnKeyCollection', () => {
    it('includes locale on keys', async () => {
      const result = await includeLocaleOnKeyCollection(projectId, {q: 'general.*', target_locale_id: '0cd76a74f8850878c9e1ac09fa22925e'})
      expect(result).toMatchInlineSnapshot(`
        Object {
          "records_affected": 6,
        }
      `)
    })
  })

  describe('excludeLocaleOnKeyCollection', () => {
    it('excludes locale on keys', async () => {
      const result = await excludeLocaleOnKeyCollection(projectId, {q: 'general.*', target_locale_id: '0cd76a74f8850878c9e1ac09fa22925e'})
      expect(result).toMatchInlineSnapshot(`
        Object {
          "records_affected": 6,
        }
      `)
    })
  })

  describe('deleteKeyCollection', () => {
    it('deletes key collection', async () => {
      const result = await deleteKeyCollection(projectId, {q: 'test.*'})
      expect(result).toMatchInlineSnapshot(`
        Object {
          "records_affected": 21,
        }
      `)
    })
  })
})
