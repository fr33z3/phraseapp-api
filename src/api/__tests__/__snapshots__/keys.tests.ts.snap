// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key api createKey creates key and return key data 1`] = `
Object {
  "comments_count": 0,
  "created_at": "2021-09-30T09:37:46Z",
  "creator": Object {
    "gravatar_uid": "f8a29c1eb4fb24f73e0cb158a4e1721b",
    "id": "2c14e21bc7e924bd7811b0d3419eebf5",
    "name": "Test Test",
    "username": "test75606",
  },
  "data_type": "string",
  "default_translation_content": null,
  "description": null,
  "format_value_type": null,
  "id": "fa22d14822b4f6bc2660d110729194bf",
  "max_characters_allowed": 0,
  "name": "test.key",
  "name_hash": "44685f0e886c174aa90dfcf48713afa0",
  "name_plural": null,
  "original_file": null,
  "plural": false,
  "screenshot_url": null,
  "tags": Array [],
  "unformatted": false,
  "updated_at": "2021-09-30T09:37:46Z",
  "xml_space_preserve": false,
}
`;

exports[`Key api getKey returns all keys for a project 1`] = `
Object {
  "comments_count": 0,
  "created_at": "2021-09-27T15:28:45Z",
  "creator": Object {
    "gravatar_uid": "f8a29c1eb4fb24f73e0cb158a4e1721b",
    "id": "2c14e21bc7e924bd7811b0d3419eebf5",
    "name": "Test Test",
    "username": "test75606",
  },
  "data_type": "string",
  "default_translation_content": "Cancel",
  "description": null,
  "format_value_type": null,
  "id": "cd0764df095295dccf8e5a0ab7c32e17",
  "max_characters_allowed": 0,
  "name": "general.cancel",
  "name_hash": "dceba40de5446d1df98a8826dc1798c6",
  "name_plural": null,
  "original_file": null,
  "plural": false,
  "screenshot_url": null,
  "tags": Array [],
  "unformatted": false,
  "updated_at": "2021-09-27T15:28:45Z",
  "xml_space_preserve": false,
}
`;

exports[`Key api listKeys returns all keys for a project 1`] = `
Array [
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "b98ed90895d651630aee37cc0c6cf94b",
    "max_characters_allowed": 0,
    "name": "general.back",
    "name_hash": "2a5fd24f37b982c1c9ff0d124e20a313",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "cd0764df095295dccf8e5a0ab7c32e17",
    "max_characters_allowed": 0,
    "name": "general.cancel",
    "name_hash": "dceba40de5446d1df98a8826dc1798c6",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "fa93069361e373ba3eb9ff2fc624b8e4",
    "max_characters_allowed": 0,
    "name": "general.confirm",
    "name_hash": "7bbbf8826f1a6f8fdb36536e97d5a991",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "497c8cd9a168801f2fff7b4dbd5d7ff5",
    "max_characters_allowed": 0,
    "name": "general.destroy",
    "name_hash": "512d0d70a4831bdf8b6fcbf3aa6e77c7",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "cff916bacfa04890d40be0029303e25e",
    "max_characters_allowed": 0,
    "name": "general.edit",
    "name_hash": "1ce842d922d18a4809f7c89b6fc268c6",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "c8b24d3f45a82a9500cb4b8a674fcab4",
    "max_characters_allowed": 0,
    "name": "general.new",
    "name_hash": "4814e003073ebc8beede68b8a5b02a3a",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "4549f6e37e7c35adfb2b43c796f4c2f4",
    "max_characters_allowed": 0,
    "name": "hello",
    "name_hash": "5d41402abc4b2a76b9719d911017c592",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "a6f26681438019f36b4376a94f35fea0",
    "max_characters_allowed": 0,
    "name": "layouts.application.about",
    "name_hash": "a29babc7e57581576d0f39f2f2c4b05e",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "418c48c108fe2cf991ee037e2519fa41",
    "max_characters_allowed": 0,
    "name": "layouts.application.account",
    "name_hash": "a287f8a2a1d3634c3609aa1210a25af7",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "efc0c8a464e3670520c00328424e5f33",
    "max_characters_allowed": 0,
    "name": "layouts.application.app_store",
    "name_hash": "f78ea636027e6e655457b0acb00d8471",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "fbe673a7707117d816faa3ee6fc20f7f",
    "max_characters_allowed": 0,
    "name": "layouts.application.imprint",
    "name_hash": "c7b1463f77a3528ac772babc2c413362",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "f1e436f0ca848860f667683aaced94d3",
    "max_characters_allowed": 0,
    "name": "layouts.application.logout",
    "name_hash": "5f76db13ae6825c50185881085fe6019",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "15a45c02eb6886ebfb21f4d71f7ed045",
    "max_characters_allowed": 0,
    "name": "layouts.application.my_mails",
    "name_hash": "843fda4ef98ea44d5548b65838fc9841",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "b81c6023fe2ff4e6e1664f3b1ad08cc0",
    "max_characters_allowed": 0,
    "name": "layouts.application.press",
    "name_hash": "6a93d34fb982019785c48c9a2c572a25",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "972a1aff4a0715e923bf9de3ceb54c6c",
    "max_characters_allowed": 0,
    "name": "layouts.application.preview",
    "name_hash": "83efa47bfa0e7088ae43781afe3f429f",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "768d4942cbf67a6211d089f47ed1549b",
    "max_characters_allowed": 0,
    "name": "layouts.application.profile",
    "name_hash": "b6e98f1f42d5a70579748640a7431ff6",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "0d36c2dc72a3018753242a1620e43e7f",
    "max_characters_allowed": 0,
    "name": "layouts.application.sign_in",
    "name_hash": "00bbee2f3f25c23c28ee8f4b1b7b51b6",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "ead7e22af0baf5faa8c607413c303bd1",
    "max_characters_allowed": 0,
    "name": "layouts.application.sign_up",
    "name_hash": "8f165881546eb5159ff8634a47704338",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
]
`;

exports[`Key api searchKeys searches keys by given query criteria 1`] = `
Array [
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "b98ed90895d651630aee37cc0c6cf94b",
    "max_characters_allowed": 0,
    "name": "general.back",
    "name_hash": "2a5fd24f37b982c1c9ff0d124e20a313",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "cd0764df095295dccf8e5a0ab7c32e17",
    "max_characters_allowed": 0,
    "name": "general.cancel",
    "name_hash": "dceba40de5446d1df98a8826dc1798c6",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "fa93069361e373ba3eb9ff2fc624b8e4",
    "max_characters_allowed": 0,
    "name": "general.confirm",
    "name_hash": "7bbbf8826f1a6f8fdb36536e97d5a991",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "497c8cd9a168801f2fff7b4dbd5d7ff5",
    "max_characters_allowed": 0,
    "name": "general.destroy",
    "name_hash": "512d0d70a4831bdf8b6fcbf3aa6e77c7",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "cff916bacfa04890d40be0029303e25e",
    "max_characters_allowed": 0,
    "name": "general.edit",
    "name_hash": "1ce842d922d18a4809f7c89b6fc268c6",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
  Object {
    "created_at": "2021-09-27T15:28:45Z",
    "description": null,
    "id": "c8b24d3f45a82a9500cb4b8a674fcab4",
    "max_characters_allowed": 0,
    "name": "general.new",
    "name_hash": "4814e003073ebc8beede68b8a5b02a3a",
    "plural": false,
    "tags": Array [],
    "updated_at": "2021-09-27T15:28:45Z",
  },
]
`;

exports[`Key api updateKey updates key and return key data 1`] = `
Object {
  "comments_count": 0,
  "created_at": "2021-09-30T09:37:46Z",
  "creator": Object {
    "gravatar_uid": "f8a29c1eb4fb24f73e0cb158a4e1721b",
    "id": "2c14e21bc7e924bd7811b0d3419eebf5",
    "name": "Test Test",
    "username": "test75606",
  },
  "data_type": "string",
  "default_translation_content": null,
  "description": null,
  "format_value_type": null,
  "id": "fa22d14822b4f6bc2660d110729194bf",
  "max_characters_allowed": 0,
  "name": "test.key-changed-name",
  "name_hash": "bea9dbcc7a09d0981ad50f5ceffb76b1",
  "name_plural": null,
  "original_file": null,
  "plural": false,
  "screenshot_url": null,
  "tags": Array [],
  "unformatted": false,
  "updated_at": "2021-09-30T09:45:14Z",
  "xml_space_preserve": false,
}
`;
